tensorboard/backend/event_processing/plugin_event_accumulator.py

```diff
@@ -310,8 +331,22 @@ def _ProcessEvent(self, event):
         if self._first_event_timestamp is None:
             self._first_event_timestamp = event.wall_time
 
+        if event.HasField("source_metadata"):
+            new_source_writer = event_util.GetSourceWriter(
+                event.source_metadata
+            )
+            if self._source_writer and self._source_writer != new_source_writer:
+                # This should not happen.
```

Comment:
I wouldn't put this quite as strongly. It's fair to say we shouldn't encounter a new source writer within the same file (since it's supposed to be only logged in the first Event). But an Accumulator is reused across multiple files within the run, and it is more likely (even if still not that common) that someone might have a case where they are using two different writers (e.g. tf.summary and jaxboard) in the same model, writing to the same log directory. In that case, we'd log this warning, but the user isn't really doing anything wrong (at least in our current approach).

I'd suggest logging at INFO instead for that reason.